src/Cloud/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,241 @@
+
+/**
+ * Wrapper for getting int64 values of a json value
+ *
+ * @param value - the json object value
+ * @param key   - the key of the wanted value as string
+ * @param error_code - the return value in case of faulty json object
+ * 
+ * @return the value or the error_code (in case of an error)
+ */
+static int64_t
+GetJsonInt(boost::json::standalone::value json_value, std::string_view key,
+           const int64_t error_code = 0) noexcept {
+  auto value = json_value.at(key);
+  return value.is_int64() ? value.to_number<int64_t>() : error_code;
+}
+
+/**
+ * Wrapper for getting string values of a json value
+ *
+ * @param value - the json object value
+ * @param key   - the key of the wanted value as string
+ * @param error_code - the return value in case of faulty json object
+ *
+ * @return the converted string value or the error_code (in case of an error)
+ */
+#ifdef _UNICODE
+static const UTF8ToWideConverter 
+GetJsonString(boost::json::standalone::value json_value, std::string_view key,
+              const char* error_code = "") noexcept {
+  auto value = json_value.at(key);
+  return UTF8ToWideConverter(value.is_string() ?
+    value.get_string().c_str() : error_code);
+}
+#else  //  _UNICODE
+static const std::string 
+GetJsonString(boost::json::standalone::value json_value, std::string_view key,
+              const char* error_code = "") noexcept {
+  auto value = json_value.at(key);
+  return value.is_string() ?
+    value.get_string().c_str() : error_code;
+}
+#endif  // _UNICODE
+
+namespace WeGlide {
+
+struct Glider {
+  uint64_t id = 0;
+  StaticString<0x40> name;
+  StaticString<4> kind;  // 'MG' - motor glider,...
+  StaticString<10> sc_class;
+};
+
+struct UploadResponse {
+  uint64_t flight_id = 0;
+  struct Pilot pilot;
+  struct Glider glider;
+  StaticString<0x40> scoring_date;
+  StaticString<0x40> registration;
+  StaticString<0x40> competition_id;
+};
+
+#if 0  // TODO this part is to delete if Codacy accept the pure function!?!
+// UpdateTask is without 'noexcept'!
+static Co::InvokeTask
+UpdateTask(Path igc_path, const char* uri, const struct Pilot& pilot,
+           uint_least32_t glider_id, boost::json::value& json,
+           ProgressListener& progress) {
+  try {
+    json = co_await UploadFlight(*Net::curl, uri, pilot, glider_id,
+      igc_path, progress);
+  }
+  catch (const std::exception& e) {
+    // Passing the exception without 'Co::InvokeTask' ???
+    throw std::runtime_error(e.what());
+  }
+}
+#else
+static Co::InvokeTask
+UpdateTask(Path igc_path, const char* uri, const struct Pilot& pilot,
+           uint_least32_t glider_id, boost::json::value& json,
+           ProgressListener& progress) {
+  json = co_await UploadFlight(*Net::curl, uri, pilot, glider_id,
+    igc_path, progress);
+}
+#endif
+
+static void
+UploadJsonInterpreter(const boost::json::value &json,
+                      struct UploadResponse& response) {
+  if (json.is_array() && (json.get_array().size() > 0)) {
```

Comment:
Never check `size()>0` - all containers have an `empty()` method instead which is faster (depending on the internal implementation).